src/IECoreImage/ImageWriter.cpp

```diff
@@ -32,96 +32,534 @@
+		}
+		else if( dataType == "int16" )
+		{
+			spec->set_format( TypeDesc::INT16 );
+		}
+		else if( dataType == "int32" )
+		{
+			spec->set_format( TypeDesc::INT32 );
+		}
+		else if( dataType == "int64" )
+		{
+			spec->set_format( TypeDesc::INT64 );
+		}
+		else if( dataType == "uint8" )
+		{
+			spec->set_format( TypeDesc::UINT8 );
+		}
+		else if( dataType == "uint16" )
+		{
+			spec->set_format( TypeDesc::UINT16 );
+		}
+		else if( dataType == "uint32" )
+		{
+			spec->set_format( TypeDesc::UINT32 );
+		}
+		else if( dataType == "uint64" )
+		{
+			spec->set_format( TypeDesc::UINT64 );
+		}
+		else if( dataType == "half" )
+		{
+			spec->set_format( TypeDesc::HALF );
+		}
+		else if( dataType == "float" )
+		{
+			spec->set_format( TypeDesc::FLOAT );
+		}
+		else if( dataType == "double" )
+		{
+			spec->set_format( TypeDesc::DOUBLE );
+		}
+	}
+
+	if( const StringData *compression = settings->member<const StringData>( "compression" ) )
+	{
+		spec->attribute( "compression", compression->readable() );
+	}
+
+	if( fileFormatName == "jpeg" )
+	{
+		spec->attribute( "CompressionQuality", settings->member<const IntData>( "quality" )->readable() );
+	}
+	else if( fileFormatName == "dpx" )
+	{
+		if( dataType == "uint10" )
+		{
+			spec->set_format( TypeDesc::UINT16 );
+			spec->attribute( "oiio:BitsPerSample", 10 );
+		}
+		else if( dataType == "uint12" )
+		{
+			spec->set_format( TypeDesc::UINT16 );
+			spec->attribute( "oiio:BitsPerSample", 12 );
+		}
+	}
+}
+
+} // namespace
+
+////////////////////////////////////////////////////////////////////////////////
+// ImageWriter
+////////////////////////////////////////////////////////////////////////////////
+
+const Writer::WriterDescription<ImageWriter> ImageWriter::g_writerDescription( OpenImageIOAlgo::extensions() );
+
+ImageWriter::ImageWriter()
+	: Writer( "Serializes images to disk using OpenImageIO", ImagePrimitiveTypeId )
 {
-	m_channelsParameter = new StringVectorParameter("channels", "The list of channels to write.  No list causes all channels to be written." );
+	constructCommon();
+}
+
+ImageWriter::ImageWriter( IECore::ObjectPtr object, const std::string &fileName )
+	: Writer( "Serializes images to disk using OpenImageIO", ImagePrimitiveTypeId )
+{
+	constructCommon();
+
+	m_objectParameter->setValue( object );
+	m_fileNameParameter->setTypedValue( fileName );
+}
+
+void ImageWriter::constructCommon()
+{
+	m_channelsParameter = new StringVectorParameter( "channels", "The list of channels to write. No list causes all channels to be written." );
+
+	m_formatSettingsParameter = new CompoundParameter( "formatSettings", "Settings specific to various file formats" );
+
+	CompoundParameterPtr exrSettings = new CompoundParameter( "openexr", "OpenEXR specific settings" );
+	m_formatSettingsParameter->addParameter( exrSettings );
+	StringParameter::PresetsContainer exrCompressionPresets;
+	exrCompressionPresets.emplace_back( StringParameter::Preset( "none", "none" ) );
```

Comment:
To be honest, I only switched to it because CLion told me to. That new stuff looks rather nice though, so I'll try that.